### PR TITLE
Feature/aapd 835

### PR DESCRIPTION
**Pull request by chrishawldev**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-835

## Description of change

Basic pipelines, docker image, package structure for the interested parties service

## Checklist

- [x] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

**Review by bryn-pins — COMMENTED** [review]
Do we need changes to deploy pipeline?

File: docker-compose.yml [hunk 1]
```diff
@@ -226,6 +226,24 @@ services:
     command: npm run start:dev
     # if enabling debug, be sure to uncomment / expose the debug port above.
     # command: npm run start:dev:debug
+  web-comment:
```
Comment (bryn-pins) [inline docker-compose.yml:229]: is it worth depending on appeals-service-api?

File: packages/web-comment/src/app.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	path.join(__dirname, './'),
+	path.join(__dirname, 'public'),
+	path.join(__dirname, 'src'),
+	path.join(__dirname, 'hello-world'),
+	path.join(__dirname, 'views')
+];
+
+nunjucks.configure(viewPaths, nunjucksConfig);
+
+if (serverConfig.server.useSecureSessionCookie) {
```
Comment (bryn-pins) [inline packages/web-comment/src/app.js:53]: think this needs defining

File: packages/web-comment/src/views/components/footer/footer.njk [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          text: "Privacy",
+          attributes: { "data-cy":"Privacy" }
+        },
+        {
+          href: "https://www.gov.uk/guidance/appeals-casework-portal-privacy-cookies",
+          text: "Cookies",
+          attributes: { "data-cy":"Cookies" }
+        },
+        {
+          href: '/accessibility-statement',
```
Comment (bryn-pins) [inline packages/web-comment/src/views/components/footer/footer.njk:35]: this page 404s think it needs bringing in

File: packages/web-comment/src/views/includes/cookie-banner.njk [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{% from "../macros/cookie-banner.njk" import cookieBanner %}
+{% from "../macros/cookie-banner-accepted.njk" import cookieBannerAccepted %}
+{% from "../macros/cookie-banner-rejected.njk" import cookieBannerRejected %}
+
+{% set showCookieBanner = displayCookieBanner %}
```
Comment (bryn-pins) [inline packages/web-comment/src/views/includes/cookie-banner.njk:5]: don't think displayCookieBanner is defined?

File: packages/web-comment/src/views/includes/body-start.njk [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{% block bodyStart %}
+
+  {% if featureFlag.googleTagManager %}
```
Comment (bryn-pins) [inline packages/web-comment/src/views/includes/body-start.njk:3]: I don't think this feature flag is defined anywhere?

**Review by bryn-pins — COMMENTED** [review]
I haven't gone through all of this but it runs locally

think there's some missing feature flag/env vars

a couple of woff2 assets 404 in the browser although get that on forms web app as well, haven't looked into it but perhaps we need most recent assets from govuk copying from node_modules? Although think mark said he was looking at a different way of achieving this

File: docker-compose.yml (hunk 1 above)
Comment (bryn-pins) [reply docker-compose.yml:229]: or perhaps can leave until we actually need to connect to api

File: packages/web-comment/src/main.njk [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+            text: "beta"
+          },
+          html: 'This is a beta service – your <a class="govuk-link" data-cy="Feedback" href="https://forms.office.com/Pages/ResponsePage.aspx?id=mN94WIhvq0iTIpmM5VcIjVqzqAxXAi1LghAWTH6Y3OJUOFg4UFdEUThGTlU3S0hFUTlERVYwMVRLTy4u">feedback</a> will help us to improve it.'
+        }) }}
+      {% endblock %}
+
+      {% block backButton %}
+        {{ govukBackLink({
+          text: "Back",
+          href: navigation[1],
```
Comment (bryn-pins) [inline packages/web-comment/src/main.njk:44]: don't think navigation is defined so this just results in an href of '#'

File: packages/web-comment/src/index.njk
Comment (bryn-pins) [inline packages/web-comment/src/index.njk:1]: seems a bit odd having index/main.njk in the root folder

does index need to exist?

could main.njk be in views folder as it'll be shared be all features?

File: packages/web-comment/src/app.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+const govukFrontEndRoot = path.join(require.resolve('govuk-frontend'), '..', '..');
+
+const viewPaths = [
+	govukFrontEndRoot,
+	// */@ministryofjustice/frontend/moj/all.js -> */@ministryofjustice/frontend
+	path.join(require.resolve('@ministryofjustice/frontend'), '..', '..'),
+	// */@pins/common/src/index.js -> */@pins/common/src/frontend
+	path.join(require.resolve('@pins/common'), '..', 'frontend'),
+	path.join(__dirname, './'),
+	path.join(__dirname, 'public'),
```
Comment (beejjacobs) [inline packages/web-comment/src/app.js:45]: Appreciate some of this is copied from the `forms-web-app`, but there won't be nunjucks in `public`. IMO we drop `public`, `src`, `hello-world`, `views`, and just include `path.join(__dirname, './')` (i.e. src directory) - then we reference everything relative to that.

**Review by bryn-pins — APPROVED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (chrishawldev) [reply docker-compose.yml:229]: Just resolving for now to get the initial merge done then we can pick these points back up

File: packages/web-comment/src/app.js (hunk 2 above)
Comment (chrishawldev) [reply packages/web-comment/src/app.js:53]: Just resolving for now to get the initial merge done then we can pick these points back up

File: packages/web-comment/src/views/components/footer/footer.njk (hunk 3 above)
Comment (chrishawldev) [reply packages/web-comment/src/views/components/footer/footer.njk:35]: Just resolving for now to get the initial merge done then we can pick these points back up

File: packages/web-comment/src/views/includes/cookie-banner.njk (hunk 4 above)
Comment (chrishawldev) [reply packages/web-comment/src/views/includes/cookie-banner.njk:5]: Just resolving for now to get the initial merge done then we can pick these points back up

File: packages/web-comment/src/views/includes/body-start.njk (hunk 5 above)
Comment (chrishawldev) [reply packages/web-comment/src/views/includes/body-start.njk:3]: Just resolving for now to get the initial merge done then we can pick these points back up

File: packages/web-comment/src/main.njk (hunk 6 above)
Comment (chrishawldev) [reply packages/web-comment/src/main.njk:44]: Just resolving for now to get the initial merge done then we can pick these points back up

File: packages/web-comment/src/index.njk
Comment (chrishawldev) [reply packages/web-comment/src/index.njk:1]: Just resolving for now to get the initial merge done then we can pick these points back up

File: packages/web-comment/src/app.js (hunk 7 above)
Comment (chrishawldev) [reply packages/web-comment/src/app.js:45]: Just resolving for now to get the initial merge done then we can pick these points back up

**Pull request merged**